### DISCUSSION TITLE
How to query account for asset types?

### DISCUSSION THREAD

**Discussion by JimAtActiv** [smartcontractkit/full-blockchain-solidity-course-py#1086] [Q&A]
Hello-

This question is a follow on to an earlier question I asked regarding how to get the balance of a particular asset associated with an account.  I was instructed to use the ERC-20 balanceOf method which solved that problem, however I now need to know how to go about getting the list of assets associated with an account .  Any suggestions?

Thanks in advance,

Jim

**Comment by cromewar** [comment]
Hello @JimAtActiv 
Not sure this is something that could be done easily with plain web3, as you should take in consideration a lot of variables.
There is a simple approach using moralis: https://www.youtube.com/watch?v=UiLI9WlNBGQ
And here you have a second approach: https://medium.com/@wbobeirne/get-all-eth-token-balances-for-multiple-addresses-in-a-single-node-call-4d0bcd1e5625
